src/index.ts

```diff
@@ -389,29 +390,32 @@ function generateShShim (src: string, to: string, opts: InternalOptions): string
   // fi
   // exit $ret
 
-  let sh = '#!/bin/sh\n'
-  sh = sh +
-    "basedir=$(dirname \"$(echo \"$0\" | sed -e 's,\\\\,/,g')\")\n" +
-    '\n' +
-    'case `uname` in\n' +
-    '    *CYGWIN*) basedir=`cygpath -w "$basedir"`;;\n' +
-    'esac\n' +
-    '\n'
+  let sh = stripIndent(`
+    #!/bin/sh
+    basedir=$(dirname "$(echo "$0" | sed -e 's,\\\\,/,g')")
+
+    case \`uname\` in
+        *CYGWIN*) basedir=\`cygpath -w "$basedir"\`;;
+    esac
+  `).trim() + '\n\n'
```

Comment:
Does it matter what EOL is used in this file? The reason this lines are joined this way is to force a specific EOL. You may see that some othe file type is joined with `\r\n`

Reply:
I just don't want to change the tests in this PR.

Reply:
I mean, if you remove the hardcoded `\n` the sh file will have `\r\n` line endings when pnpm runs on Windows. That might cause issues.

Reply:
That can be fixed by:
* Use `.editorconfig` to enforce EOL style, or
* `.split(/\r\n|\n/).join('\n')`

Reply:
a more ideal solution would be to use template literals in both source and tests, so that their EOL would either by both `CRLF` or `LF`